src/openapi/streaming/subscription.js

```diff
@@ -288,6 +288,16 @@ function onSubscribeSuccess(referenceId, result) {
  * @param response
  */
 function onSubscribeError(referenceId, response) {
+    if (response.isNetworkError) {
+        this.onUnsubscribe();
+
+        if (this.onError) {

```

Comment:
ah I see, it misses out
 if (this.queue.peekAction() !== ACTION_UNSUBSCRIBE) {
too.

what about calling onUnsubscribe at the end e.g.

if (response.isNetworkError) {
   this.onUnsubscribe();
} else {
onReadyToPerformNextAction.call(this);
}

Reply:
Done.

Need to move `setState.call(this, STATE_UNSUBSCRIBED);` as well.
But I am not sure why we are settings the state, can't we just call this.onUnsubscribe?